pkg/virt-handler/non-root.go

```diff
@@ -122,6 +122,32 @@ func (d *VirtualMachineController) prepareTap(vmi *v1.VirtualMachineInstance, re
 
 }
 
+func (*VirtualMachineController) prepareVFIO(vmi *v1.VirtualMachineInstance, res isolation.IsolationResult) error {
+	vfioPath := filepath.Join(res.MountRoot(), "dev", "vfio")
+	err := os.Chmod(filepath.Join(vfioPath, "vfio"), 0666)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+
+	groups, err := ioutil.ReadDir(vfioPath)
+	if err != nil {
+		return err
+	}
+
+	for _, group := range groups {
+		if group.Name() == "vfio" {
+			continue
+		}
+		if err := diskutils.DefaultOwnershipManager.SetFileOwnership(filepath.Join(vfioPath, group.Name())); err != nil {
+			return err
+		}
+	}
```

Comment:
Eventually what will be the mode (permissions, ownership, guid, uid) of `/dev/vfio` dir, `/dev/vfio/*` and `/dev/vfio/vfio`?

Reply:
I didn't think about `/dev/vfio`.
`/dev/vfio/vfio` will be `0666` which is enough(uid/guid doesn't matter)
`/dev/vfio/{group}` will be `qemu:qemu` and the permissions should not matter.

Reply:
I see, thanks for explaining.